core/logic/CDataPack.cpp

```
@@ -122,6 +122,34 @@ void CDataPack::PackString(const char *string)
 	elements.insert(position++, val);
 }
 
+void CDataPack::PackCellArray(cell_t const *vals, size_t count)
+{
+	InternalPack val;
+	val.type = CDataPackType::CellArray;
+	
+	if(count < 1)

```

This is failing silently but there is nothing on the path to here from a plugin to reject it, so would be an error on reading.

I think we should just allow 0-length arrays - the code here will work fine with this check removed, and it’ll just need some checks removed in the read path as well by the looks of it.